apps/coscrad-frontend/src/store/slices/notes/hooks/use-loadable-self-notes-for-resource.ts

```diff
@@ -0,0 +1,71 @@
+import {
+    IEdgeConnectionContext,
+    INoteViewModel,
+    ResourceCompositeIdentifier,
+} from '@coscrad/api-interfaces';
+import { ILoadable } from '../../interfaces/loadable.interface';
+import { compositeIdentifierMatches } from './composite-identifiers-match';
+import { useLoadableNotes } from './use-loadable-notes';
+
+export type SelfConnectionNote = {
+    id: string;
+    text: string;
+    context: IEdgeConnectionContext;
+};
+
+/**
+ * In the sake of code reuse, we may want to combine this helper
+ * with `useLoadableConnectionsToResource`. You'd pass in
+ * - a filter callback
+ * - a map callback that converts an `INoteViewModel` to the data type you need (SelfConnectionNotes here)
+ */
+export const useLoadableSelfNotesForResource = (
+    compositeIdentifierToMatch: ResourceCompositeIdentifier
+): ILoadable<SelfConnectionNote[]> => {
+    const loadableNotes = useLoadableNotes();
+
+    const { data, errorInfo, isLoading } = loadableNotes;
+
+    if (errorInfo !== null)
+        return {
+            isLoading: false,
+            errorInfo,
+            data: null,
+        };
+
+    if (isLoading || data === null)
+        return {
+            isLoading,
+            errorInfo,
+            data: null,
+        };
+
+    // The following helper is curried
+    const isTargetCompositeIdentifier = compositeIdentifierMatches(compositeIdentifierToMatch);
+
+    const selfNotesForThisResource = data
+        .filter(
+            ({ relatedResources }: INoteViewModel) =>
+                relatedResources.length === 1 &&
```

Comment:
Give the magic number a name for clarity (`NUMBER_OF_MEMBERS_IN_SELF_CONNECTION`

Reply:
Done by exposing the `connectionType` on the front-end!